connectors/sources/github.py

```diff
@@ -873,19 +884,39 @@ async def paginated_api_call(self, query, variables, keys):
     def get_repo_details(self, repo_name):
         return repo_name.split("/")
 
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=UnauthorizedException,
+    )
     async def get_personal_access_token_scopes(self):
-        request_headers = sansio.create_headers(
-            self._get_client.requester,
-            accept=sansio.accept_format(),
-            oauth_token=self._access_token(),
-        )
-        url = f"{self.base_url}/graphql"
-        _, headers, _ = await self._get_client._request("HEAD", url, request_headers)
-        scopes = headers.get("X-OAuth-Scopes")
-        if not scopes or not scopes.strip():
-            self._logger.warning(f"Couldn't find 'X-OAuth-Scopes' in headers {headers}")
-            return set()
-        return {scope.strip() for scope in scopes.split(",")}
+        try:
+            request_headers = sansio.create_headers(
+                self._get_client.requester,
+                accept=sansio.accept_format(),
+                oauth_token=self._access_token(),
+            )
+            url = f"{self.base_url}/graphql"
+            _, headers, _ = await self._get_client._request(
+                "HEAD", url, request_headers
+            )
+            scopes = headers.get("X-OAuth-Scopes")
+            if not scopes or not scopes.strip():
+                self._logger.warning(
+                    f"Couldn't find 'X-OAuth-Scopes' in headers {headers}"
+                )
+                return set()
+            return {scope.strip() for scope in scopes.split(",")}
+        except Exception as exception:
```

Comment:
Is there not a specific exception to capture here? I think it's worthwhile to separate specific error exceptions and unexpected exceptions